### ISSUE TITLE
git-hack not robust when offline

### ISSUE THREAD

**Issue by allewun** [Originate/git-town#689]
From [#664](https://github.com/Originate/git-town/issues/664#issuecomment-143747477):

> When hacking offline, all remote operations are off, so this one as well. I think that's robust and consistent, since the next git sync will sync the local repo with the remote one. 

This doesn't seem to be the case. I'm on an airplane now (spotty wifi):

``` bash
$ git hack aw-alert-polish
[dev] git fetch --prune
ssh: Could not resolve hostname github.com: nodename nor servname provided, or not known
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.

[dev] git rebase origin/dev
Current branch dev is up to date.

[dev] git checkout -b aw-alert-polish dev
Switched to a new branch 'aw-alert-polish'

[aw-alert-polish] git push -u origin aw-alert-polish
ssh: Could not resolve hostname github.com: nodename nor servname provided, or not known
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.

To abort, run "git hack --abort".
To continue after you have resolved the conflicts, run "git hack --continue".
```

**Comment by kevgo** [comment]
Hmm, yeah, sounds like we need to improve recognition of offline state!

**Comment by charlierudolph** [comment]
Closing since is related to the bash implementation and @kevgo just implemented offline mode for the go implementation. Released in 4.2.0
